2/src/com/company/TaskManagerImplements.java

```diff
@@ -0,0 +1,111 @@
+package com.company;
+
+import java.util.*;
+
+/**
+ * Created by InnaBakum on 10.11.2015.
+ */
+public class TaskManagerImplements implements TaskManager {
+
+    private Map<Date, List<Task>> task = new WeakHashMap<Date, List<Task>>();
```

Comment:
Why don't you use diamond syntax? I mean "new WeakHashMap<>()" instead of "new WeakHashMap<Date, List<Task>>()"